src/formatters/stringFormatter.js

```diff
@@ -47,3 +40,139 @@ function invalidOptionsFormatter(results) {
     return output + "\n"
   }, "\n")
 }
+
+function getSeverity(type) {
+  if (type == "error") {
+    return "error"
+  }
+
+  if (type == "info") {
+    return "info"
+  }
+
+  return "warning"
+}
+
+function logFrom(fromValue) {
+  if (fromValue.charAt(0) === "<") return fromValue
+  return path.relative(process.cwd(), fromValue).split(path.sep).join("/")
+}
+
+function getMessageWidth(columnWidths) {
+  if (!process.stdout.isTTY) {
+    return columnWidths[3]
+  }
+
+  const availableWidth = process.stdout.columns
+  const fullWidth = _.sum(_.values(columnWidths))
+
+  // If there is no reason to wrap the text, we won't align the last column to the right
+  if (availableWidth > fullWidth + marginWidths) {
+    return columnWidths[3]
+  }
+
+  return availableWidth - (fullWidth - columnWidths[3] + marginWidths)
+}
+
+function formatter(opts) {
+  const options = opts || {}
+  const sortByPosition = (typeof options.sortByPosition !== "undefined") ? options.sortByPosition : true
+  const positionless = options.positionless || "first"
+
+  return function (input) {
+    const messages = input.messages
+    const source = input.source
+
+    if (!messages.length) return ""
+
+    const orderedMessages = _.sortBy(
+      messages,
+      (m) => {
+        if (!m.line) return 1
+        if (positionless === "any") return 1
+        if (positionless === "first") return 2
+        if (positionless === "last") return 0
+      },
+      (m) => sortByPosition? m.line : 1,
+      (m) => sortByPosition? m.column : 1
+    )
+
+    // Create a list of column widths, needed to calculate
+    // the size of the message column and if needed wrap it.
+    const columnWidths = { 0: 1, 1: 1, 2: 1, 3: 1, 4: 1 }
+
+    const calculateWidths = function (columns) {
+      let width, i
+      for (i in columns) {
```

Comment:
This can be simplified using lodash's `_.forOwn` or else looping through `Object.keys`